### PR TITLE
Summary of changes for 8.12

### PR DESCRIPTION
**Pull request by mattam82**
**Kind:** documentation

Release notes for 8.12. I tried to summarize quickly the many changes in 8.12, do not hesitate to correct me if I missed something important or misinterpreted a change (in which case, I'm sorry).
Also, I didn't build it yet, maybe CI will be faster than me. That shouldn't prevent comments on the content anyway :)

### CODE REVIEW

File: doc/sphinx/changes.rst [hunk 1]
```diff
@@ -14,6 +14,91 @@ Version 8.12
...
+Coq's documentation is available at https://coq.github.io/doc/v8.12/refman (reference
+manual), and https://coq.github.io/doc/v8.12/stdlib (documentation of
+the standard library). Developer documentation of the ML API is available
+at https://coq.github.io/doc/v8.12/api.
+
+Maxime Dénès, Emilio Jesús Gallego Arias, Gaëtan Gilbert, Michael
+Soegtrop and Théo Zimmermann worked on maintaining and improving the
+continuous integration system and package building infrastructure.
+
+The OPAM repositoryy for |Coq| packages has been maintained by
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:67]:
```suggestion
The OPAM repository for |Coq| packages has been maintained by
```

File: doc/sphinx/changes.rst [hunk 2]
```diff
@@ -14,6 +14,91 @@ Version 8.12
 Summary of changes
 ~~~~~~~~~~~~~~~~~~
 
+|Coq| version 8.12 integrates many quality-of-life improvements,
+in particular with respect to notations, scopes and implicit arguments,
+along with many bug-fixes and a reworked reference manual. The main changes
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:19]:
```suggestion
along with many bug fixes and improvements to the reference manual. The main changes
```

"reworked" suggests that we're done but there's still a lot more ahead

File: doc/sphinx/changes.rst [hunk 3]
```diff
@@ -14,6 +14,91 @@ Version 8.12
...
+Li-yao, Kenji Maillard, Erik Martin-Dorel, Guillaume Melquiond, Ike Mulder,
+Guillaume Munch-Maccagnoni, Antonio Nikishaev, Karl Palmskog, Clément
+Pit-Claudel, Pierre-Marie Pédrot, Ramkumar Ramachandra, Lars Rasmusson, Daniel
+de Rauglaudre, Talia Ringer, Pierre Roux, Kazuhiko Sakaguchi, Vincent Semeria,
+@scinart, Kartik Singhal, Michael Soegtrop, Matthieu Sozeau, Enrico Tassi,
+Laurent Théry, Ralf Treinen, Anton Trunov, Bernhard M. Wiedemann, Nickolai
+Zeldovich and Théo Zimmermann.
+
+Many power users helped to improve the design of this new version via
+the issue and pull request system, the |Coq| development mailing list,
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:88]:
```suggestion
the GitHub issue and pull request system, the |Coq| development mailing list,
```

Perhaps obvious but might as well be specific as you are for the other tools

Maybe also give the email address for the dev mailing list?

File: doc/sphinx/changes.rst [hunk 4]
```diff
@@ -14,6 +14,91 @@ Version 8.12
...
+- Improvements in the treatment of implicit arguments and partially applied
+  constants in `notations`__, parsing of hexadecimal number notation and better
+  handling of scopes and coercions for printing.
+- A correct and efficient `coercion coherence`__ checking algorithm, avoiding
+  spurious or duplicate warnings.
+- An improved :g:`Search` `command`__ which accepts complex queries. Note that
+  this takes precedence over the now deprecated `ssreflect search`__.
+- Many additions and improvements of the `standard library`__.
+- Overhaul of the `reference manual`__ giving a more coherent view of the system,
+  along with consistent grammar notations that are up-to-date with Coq's grammar.
```
Comment (jfehrle) [inline doc/sphinx/changes.rst:37]:
```suggestion
- Improvements to the `reference manual`__ include a more logical organization of chapters
  along with updated syntax descriptions that match Coq's grammar in most
  but not all chapters.
```

"overhaul" incorrectly suggests we're done.

My best guess at the moment: Ltac chapter will be done within the next few days.  We may also be able to update Ltac2 and/or the tactics chapter in time for 8.12.  That leaves only SSR and several less-central chapters that don't have a lot of syntax to update.

**Review by jfehrle — COMMENTED** [review]
A few tweaks.

IIUC the announcement message for 8.12 will encourage others to help further improve the documentation.  Still a lot to be done.

**Comment by Zimmi48** [comment]
@mattam82 See c654804 (part of #12380): "[sphinx] Get rid of anonymous targets (Sphinx 2.3.1 doesn't like them)"

File: doc/sphinx/changes.rst (hunk 2 above)
Comment (Zimmi48) [reply doc/sphinx/changes.rst:19]: "improvements" is too weak to express these changes though: each version contains its lot of improvements to the refman. If you don't like "reworked" maybe "restructured" works better.

File: doc/sphinx/changes.rst [hunk 5]
```diff
@@ -14,6 +14,91 @@ Version 8.12
...
+- New notation :g:`Inductive I A | x : s := ...` to distinguish the
+  `uniform`__ from the non-uniform parameters in inductive definitions.
+- More robust and expressive treatment of `implicit inductive`__
+  parameters in inductive declarations.
+- Improvements in the treatment of implicit arguments and partially applied
+  constants in `notations`__, parsing of hexadecimal number notation and better
+  handling of scopes and coercions for printing.
+- A correct and efficient `coercion coherence`__ checking algorithm, avoiding
+  spurious or duplicate warnings.
+- An improved :g:`Search` `command`__ which accepts complex queries. Note that
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:33]:
```suggestion
- An improved :cmd:`Search` `command`__ which accepts complex queries. Note that
```

File: doc/sphinx/changes.rst (hunk 4 above)
Comment (Zimmi48) [reply doc/sphinx/changes.rst:37]: Ltac2 might be feasible for 8.12.0, but for the tactics chapter I think it is best to target 8.13.

File: doc/sphinx/changes.rst [hunk 6]
```diff
@@ -14,6 +14,91 @@ Version 8.12
...
+https://coq.inria.fr/opam/www/.
+
+The 59 contributors to this version are Abhishek Anand, Yves Bertot, Frédéric
+Besson, Lasse Blaauwbroek, Simon Boulier, Quentin Carbonneaux, Tej Chajed,
+Arthur Charguéraud, Cyril Cohen, Pierre Courtieu, Matthew Dempsky, Maxime Dénès,
+Andres Erbsen, Erika (@rrika), Nikita Eshkeev, Jim Fehrle, @formalize, Emilio
+Jesús Gallego Arias, Paolo G. Giarrusso, Gaëtan Gilbert, Jason Gross, Samuel
+Gruetter, Attila Gáspár, Hugo Herbelin, Jan-Oliver Kaiser, Robbert Krebbers,
+Vincent Laporte, Olivier Laurent, Xavier Leroy, Thomas Letan, Yishuai Li, Xia
+Li-yao, Kenji Maillard, Erik Martin-Dorel, Guillaume Melquiond, Ike Mulder,
```
Comment (Zimmi48) [inline doc/sphinx/changes.rst:79]: According to the capitalization on https://poisson.chat/, I believe "Xia" is the surname.

File: doc/sphinx/changes.rst (hunk 2 above)
Comment (jfehrle) [reply doc/sphinx/changes.rst:19]: To me, "restructured" only suggests the reorganization of sections with the new outline but not the detail changes for the syntax.  (restructure: "to change the makeup, organization, or pattern of")

How about "major improvements"?  "extensive improvements"?  "monumental improvements" :-) ?

Comment (Zimmi48) [reply doc/sphinx/changes.rst:19]: "major improvements" sounds good

File: doc/sphinx/changes.rst [hunk 7]
```diff
@@ -917,6 +1019,8 @@ Extraction
 Reference manual
 ^^^^^^^^^^^^^^^^
 
+  .. _821Refman:
```
Comment (gares) [inline doc/sphinx/changes.rst:1022]: 21?

**Comment by Zimmi48** [comment]
@mattam82 Will you have time to attend to this PR today? Do you want me to apply the requested fixes myself? We'd appreciate being able to tag as soon as tomorrow.

File: doc/sphinx/changes.rst (hunk 2 above)
Comment (mattam82) [reply doc/sphinx/changes.rst:19]: Yep, major sounds more appropriate.

**Comment by mattam82** [comment]
> @mattam82 See [c654804](https://github.com/coq/coq/commit/c654804ba038598602c1e8bdf41bf213e2c036f1) (part of #12380): "[sphinx] Get rid of anonymous targets (Sphinx 2.3.1 doesn't like them)"

Thanks, that looks much saner!

**Comment by mattam82** [comment]
> @mattam82 Will you have time to attend to this PR today? Do you want me to apply the requested fixes myself? We'd appreciate being able to tag as soon as tomorrow.

I'm on it.

File: doc/sphinx/changes.rst (hunk 7 above)
Comment (mattam82) [reply doc/sphinx/changes.rst:1022]: Fixed, thanks!

File: doc/sphinx/changes.rst (hunk 6 above)
Comment (mattam82) [reply doc/sphinx/changes.rst:79]: Right, thanks!

File: doc/sphinx/changes.rst (hunk 4 above)
Comment (mattam82) [reply doc/sphinx/changes.rst:37]: Ok. Let's not oversell it, I switch to your wording @jfehrle

**Comment by mattam82** [comment]
Actually the fix of Clément was slightly wrong, I corrected the links for 8.11 and 8.12 together.

File: doc/sphinx/changes.rst (hunk 3 above)
Comment (mattam82) [reply doc/sphinx/changes.rst]: Right, done.

**Comment by Zimmi48** [comment]
Looks like your commit message is missing newlines.

**Review by Zimmi48 — APPROVED** [review]

**Comment by mattam82** [comment]
I'll fix that now.

**Pull request merged**